packages/loader/container-loader/src/container.ts

```diff
@@ -1614,17 +1613,21 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     /**
      * Get the most recent snapshot, or a specific version.
      * @param specifiedVersion - The specific version of the snapshot to retrieve
-     * @returns The snapshot requested, or the latest snapshot if no version was specified
+     * @returns The snapshot requested, the latest snapshot if no version was specified or the oldest snapshot if
+     * version is null.
      */
-    private async fetchSnapshotTree(specifiedVersion?: string): Promise<ISnapshotTree | undefined> {
-        const version = await this.getVersion(specifiedVersion ?? this.id);
+    private async fetchSnapshotTree(specifiedVersion: string | undefined | null): Promise<ISnapshotTree | undefined> {
+        const version = await this.getVersion(specifiedVersion === undefined ? this.id : specifiedVersion);
 
         if (version !== undefined) {
             this._loadedFromVersion = version;
             return await this.storageService.getSnapshotTree(version) ?? undefined;
         } else if (specifiedVersion !== undefined) {
             // We should have a defined version to load from if specified version requested
-            this.logger.sendErrorEvent({ eventName: "NoVersionFoundWhenSpecified", specifiedVersion });
+            this.logger.sendErrorEvent({
+                eventName: "NoVersionFoundWhenSpecified",
+                specifiedVersion: specifiedVersion ?? "Oldest",
```

Comment:
this is wrong, as it looses the distinction between null and undefined. i would just log specified version directly

Reply:
Undefined is already eliminated in if condition. Can add an explicit check.

Reply:
regardless, it's best to log the actual value to explicitly understand what's happening in the code. someone could change something and then i need to correlate code version to understand what the original value was.

Reply:
> I'd go even further and use strings (i.e. some hardcoded names), but that always runs into chance of collisions :(

At least minimize the area in code where this distinction exists by immediately wrapping in an object or a union type including symbols or something.